freud/diffraction.py

```diff
@@ -500,8 +509,9 @@ def plot(self, ax=None, **kwargs):
         Returns:
             (:class:`matplotlib.axes.Axes`): Axis with the plot.
         """
-        import freud.plot
-
+        if not _HAS_MPL:
+            msg = "Plotting requires matplotlib."
+            raise ImportError(msg)
         return freud.plot.line_plot(
```

Comment:
You could define `msg` with `_HAS_MPL` so that you only have to define it once? This happens in a few files.